src/.vuepress/markdown/link.js

```diff
@@ -0,0 +1,95 @@
+// markdown-it plugin for:
+// 1. adding target="_blank" to external links
+// 2. converting internal links to <router-link>
+
+const url = require('url')
+
+const indexRE = /(^|.*\/)(index|readme).md(#?.*)$/i
+
+module.exports = (configuration) => {
+  return (md) => {
+    let hasOpenRouterLink = false
+    let hasOpenExternalLink = false
+
+    md.renderer.rules.link_open = (tokens, idx, options, env, self) => {
+    const { base } = configuration;
+    const { relativePath } = env
+      const token = tokens[idx]
+      const hrefIndex = token.attrIndex('href')
+      if (hrefIndex >= 0) {
+        const link = token.attrs[hrefIndex]
+        const href = link[1]
+        const isExternal = /^https?:/.test(href)
+        const startsWithBaseURL = href.startsWith(base)
+        
+        if (isExternal) {
+          hasOpenExternalLink = true
+          token.attrSet('target', '_blank')
+          token.attrSet('rel', 'noopener noreferrer')
+        } else if (startsWithBaseURL) {
+          hasOpenRouterLink = true
+          tokens[idx] = toRouterLink(token, link, relativePath, base)
+        }
+      }
+      return self.renderToken(tokens, idx, options)
+    }
+
+    function toRouterLink(token, link, relativePath, base) {
+      link[0] = 'to'
+      let to = link[1].replace(base, '')
+
+      // convert link to filename and export it for existence check
+      const links = md.$data.links || (md.$data.links = [])
+      links.push(to)
+
+      // relative path usage.
+      if (!to.startsWith('/')) {
+        to = relativePath
+          ? url.resolve('/' + relativePath, to)
+          : ensureBeginningDotSlash(to)
+      }
+
+      const indexMatch = to.match(indexRE)
+      if (indexMatch) {
+        const [, path, , hash] = indexMatch
+        to = path + hash
+      } else {
+        to = to
+          .replace(/\.md$/, '.html')
+          .replace(/\.md(#.*)$/, '.html$1')
```

Comment:
We should support URLs written like this: `http://foo/bar/baz.md/#qux`

```suggestion
          .replace(/\.md\/?(#.*)$/, '.html$1')
```

Reply:
Er... Why?